### DISCUSSION TITLE
Setting up hierarchies

### DISCUSSION THREAD

**Discussion by hieu6** [SkepticMystic/breadcrumbs#106] [Q&A]
hello friends! thank you so much @SkepticMystic for your contributions to the community; you've had the answer to most questions i've had when googling around re: dataview/templater questions this past week. appreciate you!

i am unsure if i have setup the breadcrumb hierarchy correctly. 
right now for my `project management` vault, i have the hierarchy setup like this, from highest to lowest hierarchy. there are no "siblings" in this hierarchy

```
pillars:: 
value-goals:: 
goal-outcomes:: 
projects:: 
actions:: 
```
### Q: is this how it should be setup in the breadcrumb plugin settings?

![Screen Shot 2021-08-22 at 11 42 30](https://user-images.githubusercontent.com/72253321/130366367-0900b1e7-02bc-47aa-935f-02f9ce8f3749.png)

---

### Q: could you explain to me how i would fill this out, using the hierarchy examples above?

![Screen Shot 2021-08-22 at 11 46 05](https://user-images.githubusercontent.com/72253321/130366478-dddd536d-006b-4215-9255-df8102cba306.png)

---

visual examples, screenshots, or .md files of example hierarchy setups and how it would look like displayed would be very helpful for my understanding! thank you again @SkepticMystic for your helpful explanations. they've made the more intimidating tools of obsidian much more accessible to me. :D :D

**Comment by SkepticMystic** [comment]
Hey @hieu6, thanks for the detailed issue 🙂 
You've caught me right before going to bed, so I won't be able to stay long. Hopefully the issue is simple, but I am happy to help out more fully tomorrow!
I think the issue might be as simple as the way you have the hierarchies set up. I think you just need to take out the arrows in the fields, so inside the text field, _just_ give the relationship name. So use `pillars` instead of `↑pillars`, you know what I mean? Does that fix the issue? Be sure to Refresh Index after making the change and saving the heirarchies.

This has been an issue a couple times, so in the latest release, I've changed it so that the arrows aren't actual values in the input box, just a placeholder.

Let me know if this works, otherwise I'll be able to help more tomorrow 😄

**Reply by SkepticMystic** [reply]
@hieu6 did you come right with your first question?

**Reply by hieu6** [reply]
thanks for checking in! I did get it to work, thanks to the recent update you mentioned. breadcrumbs saved me when I got lost setting up a new vault with a PPV like structure.

there seemed to be so much overhead though, setting up all the metadata. I'm sure it seemed more intense cuz I was setting up a few dozen note hubs at once. 

after a while, I was unsure if the breadcrumb specific hierarchy was worth it, bc thanks to the good habits it ingrained into me, I'd have the inline dataview links to top/bottom structures anyway.

what is your note structure setup workflow like? how do you not get overwhelmed by micromanaging the odds and ends? 

the preview trail is super cute and I'd love to keep it. I think if I can streamline my flow or maybe not be as strict for getting them in for all the notes.... _all the notes are important in my heart though_

**Reply by SkepticMystic** [reply]
I hear you, there is definitely alot of set up and maintenance of the hierarchy.
I don't see an easy way around this though because I can't predict the structure, you know? Breadcrumbs needs that info to work.

Personally, I haven't had that much hassle setting up my hierarchy. I have about 1500 notes, so nothing crazy, but still a few. It took a couple manhours to get things all set up, but now it's done. If I have a new note to add to the hierarchy, I'll do that when I create the note.
So I guess my approach is to do little pieces, often.

Do you have suggestions on how this could be streamlined, or even done automatically?

**Comment by SkepticMystic** [comment]
Regarding your second question, maybe the wiki page on [Hierarchy Notes](https://github.com/SkepticMystic/breadcrumbs/wiki/Hierarchy-Notes) is enough to answer your question. If not, I'll give more detail tomorrow!

**Reply by hieu6** [reply]
oh wow, you have a great wiki. this is so cool. THANK YOU that's very helpful.

**Comment by hieu6** [comment]
oh it wasn't a critique on the plugin! noticing that it's a mindset or
workflow shift. wanted to see if perhaps i was missing a better way. i've
added the hierarchies as a text expansion, is a natural part of workflow
now. thank you again for your work!



hieu
they/them

**Reply by SkepticMystic** [reply]
Hey hieu :)
Haha I'm also looking for a better way to do things! I guess it's an open question then, if there is some way to automatically infer the structure. But yes, for now it is quite manual.

**Comment by hieu6** [comment]
the fear runs so deep...
*what if I forget to link this one and it's lost forever!*

personally a big issue for me is my memory is so bad, I'll instantly forget
where I just clicked this link from. maybe i should utilize the recent
files one too.

I use some variation of Johnny decimal for my project files, say 2.1 for
objective, then 2.1a, b, c etc. for projects, further down for project
resources... could there be some variation of hierarchy notes where we
could set up a similar naming convention to be recognized as hierarchy? I'm
a pleb idk how the magic works but here's extra fairy dust ✨ *I'm helping*

**Reply by SkepticMystic** [reply]
Interesting... I think I could work with that.
It seems too simple... haha maybe it is? Using the numbering convention, you could rebuild the hierarchy from the note names

**Comment by hieu6** [comment]
👀!

here's an example for INSPO

![Screen Shot 2021-09-01 at 02.17.14.png](https://user-images.githubusercontent.com/72253321/131646401-0e5c5363-6b26-4a21-b808-43bfc6260cb3.png)

**Reply by SkepticMystic** [reply]
Thanks, that's helpful!
One problem I see is that here:

![image](https://user-images.githubusercontent.com/70717676/131646673-7693f82a-5fe5-402a-89b6-22a6da411968.png)

You would have to name the last few notes as `2.1.a.1` (not `2.1.a1`). Atleast the way I am thinking of implementing it.
My approach is to just split the numbering at `.` to get `[2, 1, a, 1]`. So having `[2, 1, a1]` would be different, you know what I mean?

**Reply by hieu6** [reply]
first, oh god so sorry for the ugly replies. these were via email. 

on topic, if you do end up implementing the above structure, would it matter if it's a dash or a period? say `2.1.B.1-1`

**Reply by SkepticMystic** [reply]
Hahahaha it is a bit of a mess here 😂 
But that's the price you pay for innovation 😋 
Ummmm I could make it a setting. So you'd be able to say where to split the numbering. I _suppose_ you could give more than one delimiter... I'll have to workshop it. Can you add this to your FR :)

**Reply by SkepticMystic** [reply]
So in your example, Breadcrumbs would see the structure as `[2, 1, B, 1, 1]`

**Comment by hieu6** [comment]
oh ya, there's no rhyme or reason for sudden changes. would be so so happy
to formalize it, I welcome the excuse to fiddle with it. oh man I'm so
excited xD

**Reply by SkepticMystic** [reply]
Hahahaha ok cool.
Just to be sure, you can use whatever numbering you want, it just needs to agree with it's parents.

So having `2.LongTitle` → `2.LongTitle.Shorter` is fine. But the example above wouldn't be

**Reply by SkepticMystic** [reply]
Can you please add a dedicated FR for this 😄

**Comment by hieu6** [comment]
boom, I tried my best 😌

https://github.com/SkepticMystic/breadcrumbs/issues/122
